src/lib.rs

```diff
@@ -14,39 +24,184 @@ fn process_instruction(
+                    &spl_token::id(),
+                ),
+                &[
+                    funder_info.clone(),
+                    program_token_info.clone(),
+                    system_program_info.clone(),
+                ],
+                &[&program_token_signer_seeds],
+            )?;
+
+            info!("Initializing program token account");
+            invoke(
+                &spl_token::instruction::initialize_account(
+                    &spl_token::id(),
+                    program_token_info.key,
+                    mint_info.key,
+                    program_token_info.key, // token owner is also `program_token` address
+                )?,
+                &[
+                    program_token_info.clone(),
+                    spl_token_program_info.clone(),
+                    rent_sysvar_info.clone(),
+                    mint_info.clone(),
+                ],
+            )?;
+            Ok(())
+        }
+        Some(1) => {
+            info!("Close program token account...");
+            let funder_info = next_account_info(account_info_iter)?;
+            let spl_token_program_info = next_account_info(account_info_iter)?;
+
+            invoke_signed(
+                &spl_token::instruction::close_account(
+                    &spl_token::id(),
+                    program_token_info.key,
+                    funder_info.key,
+                    program_token_info.key, // token owner is also `program_token` address
+                    &[],
+                )
+                .expect("close_account"),
+                &[
+                    funder_info.clone(),
+                    spl_token_program_info.clone(),
+                    program_token_info.clone(),
+                ],
+                &[&program_token_signer_seeds],
+            )
+        }
+        _ => {
+            info!("Error: Unsupported instruction");
+            Err(ProgramError::InvalidInstructionData)
+        }
+    }
 }
 
 #[cfg(test)]
 mod test {
+    #![cfg(feature = "test-bpf")]
+
     use super::*;
     use assert_matches::*;
-    use solana_program::instruction::{AccountMeta, Instruction};
+    use solana_program::{
+        instruction::{AccountMeta, Instruction},
+        sysvar,
+    };
     use solana_program_test::*;
     use solana_sdk::{signature::Signer, transaction::Transaction};
 
+    fn program_test(program_id: Pubkey) -> ProgramTest {
+        let mut pc = ProgramTest::new(
+            "bpf_program_template",
+            program_id,
+            processor!(process_instruction),
+        );
+
+        // Add SPL Token program
+        pc.add_program(
+            "spl_token",
+            spl_token::id(),
+            processor!(spl_token::processor::Processor::process),
+        );
+
+        pc
+    }
+
     #[tokio::test]
-    async fn test_transaction() {
+    async fn test_create_then_close() {
         let program_id = Pubkey::new_unique();
```

Comment:
why is a `program_id` created here, when we have the `pc.add_program()` call in `fn program_test()`, which should just create a mock spl_token to use?

Reply:
i was just interpreting it incorrectly, `Pubkey::new_unique()` creates the program_id we then instantiante

Reply:
This is the `program_id for the "bpf-program-template program, passed in via `ProgramTest::new(..)`.  It's just a unnamed, unique id here because the actual value doesn't matter for testing purposes -- it just needs to be something.   On a real cluster, this would be the value returned by `solana deploy` when the program is deployed